persistent/cPickleCache.c

```diff
@@ -636,17 +641,31 @@ cc_oid_unreferenced(ccobject *self, PyObject *oid)
     Py_INCREF(dead_pers_obj);
 
     if (PyDict_DelItem(self->data, oid) < 0)
-        return;
+    {
+        /* Almost ignore errors if it wasn't already present (somehow;
+           that shouldn't be possible since we literally just got it out
+           of this dict and we're holding the GIL and not making any
+           calls that could cause a greenlet switch so the state of the
+           dictionary should not change). We still need to finish the cleanup.
+
+           Just write an unraisable error (like an exception from __del__,
+           because that's basically what this is).
+        */
+        PyErr_WriteUnraisable((PyObject*)dead_pers_obj);
+        PyErr_Clear();
+	/* Have the same side effect of clearing a ref count as the dict would have.*/
+	Py_DECREF(dead_pers_obj);
```

Comment:
The indentation seems off here.

Reply:
Thanks. That file has a mix of tabs and spaces for indentation and it confuses my editor.